### ISSUE TITLE
Static Decode Wrapper Design for Llama Decoder Layer

### ISSUE THREAD

**Issue by mhs4670go** [Samsung/TICO#507]
## Background

The [prefill wrapper](https://github.com/Samsung/TICO/blob/216f2c47393eb87f2e86b194767439ba1ed13402/tico/quantization/wrapq/wrappers/llama/quant_decoder_layer.py) is already implemented with fully static shapes.

We now need a decode wrapper that also satisfies strict static shape requirements while supporting KV cache updates.

##  Goal

Design a decode wrapper with:

- Fully static tensor shapes
- No dynamic mask construction inside forward
- Correct RoPE positional behavior

### Key Idea

The decode model always assumes:
```
total_kv_length = max_seq_len
```

The KV cache input represents:

```
[past tokens | padding] or [padding | past tokens]
```

The current token (seq=1) is appended internally during attention computation.

**Padding regions are masked via the attention mask.**

### Input Specification
```
x: (1, 1, model_dim)
# RoPE embeddings for the current token
position_embeddings:
  cos: (1, 1, head_dim)
  sin: (1, 1, head_dim)
attention_mask: (1, 1, max_seq_len)
past_key:   (1, num_kv_heads, max_seq_len - 1, head_dim)
past_value: (1, num_kv_heads, max_seq_len - 1, head_dim)
```

### Output Specification
```
out: (1, 1, model_dim)
# New KV entries (delta only):
new_key:   (1, num_kv_heads, 1, head_dim)
new_value: (1, num_kv_heads, 1, head_dim)
```

The host runtime is responsible for updating the external KV cache buffer.
